src/plots/cartesian/type_defaults.js

```diff
@@ -89,9 +89,10 @@ function setAutoType(ax, data) {
     }
     else if(d0.type === 'splom') {
         var dimensions = d0.dimensions;
+        var diag = d0._diag;
```

Comment:
Recall that `_diag` is:

https://github.com/plotly/plotly.js/blob/def6aa5a24527c68e90f50485255170609640e43/src/traces/splom/defaults.js#L104-L124

Reply:
ah ok - a little confusing as usually `xa` is a full axis object, an id would be `xId` or something... but anyway nice fix!

Reply:
cleaned up in -> https://github.com/plotly/plotly.js/pull/2899/commits/f3c73dbd774f2280b7e0cdfee0d4170e158b889d